### PR TITLE
Replace "dynamically injects bytecode" in Python zero-code instrumentation

### PR DESCRIPTION
**Pull request by CFly17**
Update/clarify the automatic instrumentation method for Python zero-code instrumentation

- replace "dynamically injects bytecode" in Introduction section with a brief description of monkey patching
- add details to Setup section about how the `opentelemetry-bootstrap -a install` command uses monkey patching

### CODE REVIEW

**Review by cartermp — APPROVED** [review]

**Comment by cartermp** [comment]
@open-telemetry/python-approvers PTAL!

File: content/en/docs/zero-code/python/_index.md
Comment (xrmx) [inline content/en/docs/zero-code/python/_index.md:43]: I would like to wikipedia article about monkey patching instead. We are not using `monkeypatch`.

**Review by xrmx — CHANGES_REQUESTED** [review]

File: content/en/docs/zero-code/python/_index.md
Comment (emdneto) [reply content/en/docs/zero-code/python/_index.md:43]: Also, I would add the reference to the first time where it was mentioned

File: static/refcache.json
Comment (emdneto) [inline static/refcache.json:1714]: Should update this too?

**Review by xrmx — APPROVED** [review]
With the refcache fixed LGTM

File: static/refcache.json
Comment (xrmx) [inline static/refcache.json:1714]: I think this chunk is not needed, looks like fixrefcache only adds entries

Comment (CFly17) [reply static/refcache.json:1714]: Thanks for the review @xrmx. It means a lot; this is my first contribution to open source. 

I was unfamiliar with comment-triggered actions like `/fix:refcache`. Should only maintainers do this?

I am also reading the [contributing process](https://github.com/open-telemetry/community/blob/main/guides/contributor/processes.md) and wonder if there a 'checklist' to clean up my local before committing, e.g. `npx prettier --write`?

Comment (svrnm) [reply static/refcache.json:1714]: Hey @CFly17! Thanks for your contribution, happy to read that you choose otel for being your first!

The `/fix:*` commands are just automations we use run the `npm fix:*` commands without pulling your PR, running it locally and pusing it back. You may want to run `npm fix:all` before pushing your PR, but since we sometimes do inline changes, etc. we have no issues with a PR that has failing CI checks. The thing that counts is that those checks are green shortly before merging the PR, and this can either be done by you or by a maintainer. 

Note, that this is how we (OTel Docs Maintainer) are doing it, other OSS projects might choose a different approach

**Pull request merged**
